android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -2266,21 +2283,80 @@ public static int getBannerHeight(Context context) {
     } else if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
-      return prefs.getInt(KMManager.KMKey_KeyboardHeightLandscape, defaultHeight);
+        return KMManager.KeyboardHeight_Context_Landscape_Current;
+    } else {
+      return KMManager.KeyboardHeight_Invalid; // Invalid orientation
     }
-
-    return defaultHeight;
   }
 
+  /**
+   * Apply the keyboard height for the current orientation and store it in SharedPreferences.
+   * If height is KMManager.KeyboardHeight_Reset, remove the stored height for that 
+   * orientation, resetting to the default height for the device/orientation.
+   * @param context Context
+   * @param height int - keyboard height in dp
+   */
   public static void applyKeyboardHeight(Context context, int height) {
+    int orientation = getOrientation(context);
+    applyKeyboardHeight(context, height, orientation); 
+  }
+
+  /**
+   * Apply the keyboard height for the specified orientation and store it in SharedPreferences.
+   * If height is KMManager.KeyboardHeight_Reset, remove the stored height for that orientation.
+   * @param context Context
+   * @param height int - keyboard height in dp
+   * @param orientation int - Configuration.ORIENTATION_PORTRAIT or Configuration.ORIENTATION_LANDSCAPE
+   */
+  public static void applyKeyboardHeight(Context context, int height, int orientation) {
+    SharedPreferences prefs = context.getSharedPreferences(KMManager.KMEngine_PrefsKey, Context.MODE_PRIVATE);
+    SharedPreferences.Editor editor = prefs.edit();
+    int defaultHeightForContext = getDefaultKeyboardHeight(orientation);
+    if (orientation != Configuration.ORIENTATION_PORTRAIT && orientation != Configuration.ORIENTATION_LANDSCAPE) {
+      return; // Invalid orientation, do nothing
+    }
+    if (height == KMManager.KeyboardHeight_Reset) {
+      // Passing 0 will reset the stored height for this orientation.
+      if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
+        editor.remove(KMManager.KMKey_KeyboardHeightLandscape);
+      } else /* (orientation == Configuration.ORIENTATION_PORTRAIT) */{
+        editor.remove(KMManager.KMKey_KeyboardHeightPortrait);
+      } 
+      height = defaultHeightForContext;
+    } else {
+      // Applying gating to 50%-200% of default height (following Keyman) 
+      if (height < (defaultHeightForContext / 2)) {
+        height = (int) (defaultHeightForContext / 2);
+      } else if (height > (defaultHeightForContext * 2)) {
+        height = (int) (defaultHeightForContext * 2);
+      } 
+
+      // Store the new height based on the current orientation
+      if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
+        editor.putInt(KMManager.KMKey_KeyboardHeightLandscape, height);
+      } else /* (orientation == Configuration.ORIENTATION_PORTRAIT) */ {
+        editor.putInt(KMManager.KMKey_KeyboardHeightPortrait, height);
+      } 
+    }
+    editor.commit();
+    // Confirm new LayoutParams for in-app or system keyboards 
```

Comment:
We need to set KMManager.KeyboardHeight_Context_Portrait_Current / KMManager.KeyboardHeight_Context_Landscape_Current here. Missed this in the earlier PR.

Reply:
Done. So many orientation checks :disappointed:

Reply:
I guess we can let this fix filter back into alpha through the regular beta->alpha PR.